src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/TseViewApplicationHelper.java

```diff
@@ -0,0 +1,283 @@
+                                ? "N/A" : response.getValue().getCopyNoGiveDetails()));
+    }
+
+    private static String createResponseTable(List<TseRespondTypeItem> respondList, String applicant) {
+        AtomicInteger count = new AtomicInteger(0);
+        return RESPONSE_LIST_TITLE + respondList.stream().map((TseRespondTypeItem response) -> {
+            count.getAndIncrement();
+            if (ADMIN.equals(response.getValue().getFrom())) {
+                return createAdminResponse(response, count);
+            } else {
+                return createRespondentOrClaimantResponse(response, count, applicant);
+            }
+        }).collect(Collectors.joining(""));
+    }
+
+    /**
+     * Set the markup for an application summary and a table of responses
+     * for the 'view an application' event.
+     * @param caseData
+     */
+    public static void setDataForTseApplicationSummaryAndResponses(CaseData caseData) {
+        List<GenericTseApplicationTypeItem> applications = caseData.getGenericTseApplicationCollection();
+        if (CollectionUtils.isEmpty(applications) || getChosenApplication(caseData) == null) {
+            return;
+        }
+        GenericTseApplicationType genericTseApplicationType = getChosenApplication(caseData);
+        String document = "N/A";
+        if (genericTseApplicationType.getDocumentUpload() != null) {
+            Pattern pattern = Pattern.compile("^.+?/documents/");
+            Matcher matcher = pattern.matcher(genericTseApplicationType.getDocumentUpload().getDocumentBinaryUrl());
+            String documentLink = matcher.replaceFirst("");
+            String documentName = genericTseApplicationType.getDocumentUpload().getDocumentFilename();
+            document = String.format("<a href=\"/documents/%s\" target=\"_blank\">%s</a>", documentLink, documentName);
+        }
+        String respondTablesCollection = "";
+        if (!CollectionUtils.isEmpty(genericTseApplicationType.getRespondCollection())) {
+            List<TseRespondTypeItem> respondList = genericTseApplicationType.getRespondCollection();
+            respondTablesCollection = createResponseTable(respondList, genericTseApplicationType.getApplicant());
+        }
+        caseData.setTseApplicationSummaryAndResponsesMarkup(
+                String.format(
+                        APPLICATION_DETAILS, genericTseApplicationType.getApplicant(),
+                        genericTseApplicationType.getType(),
+                        genericTseApplicationType.getDate(),
+                        isNullOrEmpty(genericTseApplicationType.getDetails()) ? "N/A"
+                                : genericTseApplicationType.getDetails(),
+                        document,
+                        isNullOrEmpty(genericTseApplicationType.getCopyToOtherPartyYesOrNo()) ? "N/A"
+                                : genericTseApplicationType.getCopyToOtherPartyYesOrNo()
+                ) + respondTablesCollection);
```

Comment:
String.format + concatenation is a bit messy, can this be included on APPLICATION_DETAILS?